test/extended/util/client.go

```diff
@@ -111,62 +109,63 @@ type resourceRef struct {
 	Name      string
 }
 
-// NewCLIWithFramework initializes the CLI using the provided Kube
-// framework. It can be called inside of a Ginkgo .It() function.
-func NewCLIWithFramework(kubeFramework *framework.Framework) *CLI {
-	cli := &CLI{
-		kubeFramework:           kubeFramework,
-		username:                "admin",
-		execPath:                "oc",
-		adminConfigPath:         KubeConfigPath(),
-		staticConfigManifestDir: StaticConfigManifestDir(),
+// CliOptions allows customization of the newely created CLI resource
+type CliOptions struct {
+	// KubeFramework to use, instead of creating a brand new one
+	KubeFramework *framework.Framework
+	// prefix for the created test namespace
+	BaseName string
```

Comment:
seems like this and withoutnamespace should be mutually exclusive.

Reply:
Do we really need to set BaseName when withoutnamespace is true?

Reply:
BaseName is used as a UniqueName here: https://github.com/kubernetes/kubernetes/blob/24fbb13eafec665e6f4b64961930774babd8b6d1/test/e2e/framework/framework.go#L279-L282 we could not require it being set and allow it to be randomly generated when empty, especially that this `f.UniqueName` is used pretty heavily in some cases. So for now I'd prefer we just keep it in there :wink: